trino-rest-github/src/main/java/pl/net/was/rest/github/Sync.java

```diff
@@ -1852,13 +1855,29 @@ private static int getEmptyLimit()
         return Integer.parseInt(limit);
     }
 
+    private static long getQueryDelayMillis()
+    {
+        String queryDelay = System.getenv("QUERY_DELAY_MS");
+        if (queryDelay == null) {
+            return 0L;
+        }
+        return Long.parseLong(queryDelay);
+    }
+
     private static int retryExecute(PreparedStatement statement)
             throws SQLException
     {
         int breaker = 3;
+        long queryDelay = getQueryDelayMillis();
         while (true) {
+            long startTime = System.currentTimeMillis();
+
             try {
-                return statement.executeUpdate();
+                int queryResult = statement.executeUpdate();
+                if (queryDelay > 0) {
```

Comment:
This condition is unnecessary: 
> Parameters:
>    timeout - the minimum time to sleep. If less than or equal to zero, do not sleep at all.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/TimeUnit.html#sleep(long)

Reply:
Good catch! 🥇